collect/system-info.js

```diff
@@ -2,7 +2,14 @@
 
 const path = require('path')
 const Module = require('module')
-const asyncWrap = process.binding('async_wrap') // eslint-disable-line node/no-deprecated-api
+const semver = require('semver')
+let asyncWrap
+
+if (semver.gte(process.version, '17.2.0')) {
```

Comment:
instead of relying on the Node.js version, just detect if `require('async_hooks').asyncWrapProviders` is defined